packages/clay-dropdown/src/ClayDropdownBase.js

```diff
@@ -121,6 +140,165 @@ class ClayDropdownBase extends ClayComponent {
 		});
 	}
 
+	/**
+	 * Increments to the index to the next item.
+	 * @param {Array} items
+	 * @param {Number} active
+	 * @param {Number} subActive
+	 * @param {Boolean} reverse
+	 * @private
+	 * @return {Object}
+	 */
+	_incrementActiveItemIndex(items, active, subActive, reverse) {
+		const totalItems = items.length;
+
+		subActive = reverse ? subActive - 1 : subActive + 1;
+
+		if (reverse) {
+			if (subActive < 0) {
+				const nextActive = active - 1;
+
+				active = nextActive < 0 ? totalItems - 1 : nextActive;
+
+				subActive = items[active].length - 1;
+			}
+		} else {
+			if (subActive === items[active].length) {
+				const nextActive = active + 1;
+
+				active = nextActive === totalItems ? 0 : nextActive;
+
+				subActive = 0;
+			}
+		}
+
+		return {active, subActive};
+	}
+
+	/**
+	 * Gets the indexes of the next active item.
+	 * @param {Array} items
+	 * @param {Number} initialActive
+	 * @param {Number} initialSubActive
+	 * @param {Boolean} reverse
+	 * @private
+	 * @return {Number}
+	 */
+	_getNextIndexes(items, initialActive, initialSubActive, reverse = false) {
+		let {active, subActive} = this._incrementActiveItemIndex(
+			items,
+			initialActive === -1 ? 0 : initialActive,
+			initialSubActive,
+			reverse
+		);
+
+		const totalItems = items.flat(2);
```

Comment:
I'm just a bit worried about using `flat` here, it's not supported in some browsers yet and as we support IE 11 for now 🙂... maybe we need a polyfill in Portal, I'm not sure we already have it there. @carloslancha, @julien or @jbalsas do you know anything about this?

Reply:
Oh good catch. I didn't mean to leave that in there.